primedev/mods/modmanager.cpp

```diff
@@ -866,27 +866,57 @@ void ModManager::LoadMods()
 				if (fs::is_regular_file(file) && file.path().extension() == ".rpak")
 				{
 					std::string pakName(file.path().filename().string());
-					ModRpakEntry& modPak = mod.Rpaks.emplace_back();
+					ModRpakEntry& modPak = mod.Rpaks.emplace_back(mod);
+
+					modPak.m_pakName = pakName;
 
 					if (!bUseRpakJson)
 					{
 						spdlog::warn("Mod {} contains rpaks without valid rpak.json, rpaks might not be loaded", mod.Name);
 					}
 					else
 					{
-						modPak.m_bAutoLoad =
+						modPak.m_preload =
 							(dRpakJson.HasMember("Preload") && dRpakJson["Preload"].IsObject() && dRpakJson["Preload"].HasMember(pakName) &&
 							 dRpakJson["Preload"][pakName].IsTrue());
 
+						// only one load method can be used for an rpak.
+						if (modPak.m_preload)
+							goto REGISTER_STARPAK;
```

Comment:
Don't use `goto` we're trying to do semi-modern c++ not c

Reply:
`goto` was the cleanest way that I could think of to write this logic without splitting up this monolithic function (which I would love to do but that's for a different PR I think)

Reply:
Isn't "refactor" in the name of this PR? Why wait :)

Reply:
Me when #171